src/main/java/games/strategy/triplea/ui/PlacePanel.java

```diff
@@ -88,9 +88,7 @@ public void territorySelected(final Territory territory, final MouseDetails e) {
       final UnitChooser chooser = new UnitChooser(units, Collections.emptyMap(), getData(),
           false, getMap().getUIContext());
       final String messageText = "Place units in " + territory.getName();
-      if (maxUnits[0] > 0) {
-        chooser.setMaxAndShowMaxButton(maxUnits[0]);
-      }
+      chooser.setMaxAndShowMaxButton(maxUnits[0]);
```

Comment:
Yeah I think >= 0 would be better here as maxUnits can be -1 if its unlimited and we probably don't want to show the max button in that case.